examples/pipeline_wavernn/wavernn.py

```diff
@@ -0,0 +1,586 @@
+    # Install signal handler
+    signal.signal(signal.SIGUSR1, lambda a, b: signal_handler(a, b))
+
+    # use torchaudio transform to get waveform and specgram
+
+#    melkwargs = {
+#        "n_fft": 2048,
+#        "n_mels": args.n_freq,
+#        "hop_length": args.hop_length,
+#        "f_min": args.f_min,
+#        "win_length": args.win_length
+#    }
+
+#    transforms = torch.nn.Sequential(
+#        torchaudio.transforms.MelSpectrogram(
+#            sample_rate=args.sample_rate, **melkwargs
+#        ),
+#    )
+
+    # use librosa transform to get waveform and specgram
+
+    melkwargs = {
+        "n_fft": 2048,
+        "num_mels": args.n_freq,
+        "hop_length": args.hop_length,
+        "fmin": args.f_min,
+        "win_length": args.win_length,
+        "sample_rate": args.sample_rate,
+        "min_level_db": args.min_level_db
+    }
+    transforms = Transform(**melkwargs)
+
+    # dataset
+    train_dataset, test_dataset = datasets_ljspeech(args, transforms)
+
+    loader_training_params = {
+        "num_workers": args.workers,
+        "pin_memory": False,
+        "shuffle": True,
+        "drop_last": False,
+    }
+    loader_validation_params = loader_training_params.copy()
+    loader_validation_params["shuffle"] = False
+
+    collate_fn = collate_factory(args)
+
+    loader_training = DataLoader(
+        train_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_training_params,
+    )
+    loader_test = DataLoader(
+        test_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_validation_params,
+    )
+
+    # model
+    model = _WaveRNN(upsample_scales=args.upsample_scales,
+                     n_bits=args.n_bits,
+                     sample_rate=args.sample_rate,
+                     hop_length=args.hop_length,
+                     n_res_block=args.n_res_block,
+                     n_rnn=args.n_rnn,
+                     n_fc=args.n_fc,
+                     kernel_size=args.kernel_size,
+                     n_freq=args.n_freq,
+                     n_hidden=args.n_hidden,
+                     n_output=args.n_output,
+                     mode=args.mode,
+                     )
+
+    if args.jit:
+        model = torch.jit.script(model)
+
+    model = torch.nn.DataParallel(model)
+    model = model.to(devices[0], non_blocking=True)
+
+    n = count_parameters(model)
+    logging.info(f"Number of parameters: {n}")
+
+    # Optimizer
+    optimizer_params = {
+        "lr": args.learning_rate,
+        "betas": (args.adam_beta1, args.adam_beta2),
+        "eps": args.eps,
+        "weight_decay": args.weight_decay,
+    }
+
+    optimizer = Adam(model.parameters(), **optimizer_params)
+
+    if args.scheduler == "exponential":
+        scheduler = ExponentialLR(optimizer, gamma=args.gamma)
+    elif args.scheduler == "reduceonplateau":
+        scheduler = ReduceLROnPlateau(optimizer, patience=10, threshold=1e-3)
```

Comment:
Have you tried your code with this scheduler? It will generate an error, since `scheduler.step` needs a parameter :)

Reply:
I didn't try this scheduler. Now this part is removed.